prow/gangway/gangway.go

```diff
@@ -0,0 +1,740 @@
+			SHA: refToMerge.GetCommit(),
+			Ref: refToMerge.GetRef(),
+		}
+		pulls = append(pulls, pull)
+	}
+
+	refs.Pulls = pulls
+
+	// FIXME: allow API callers to supply extra repos to clone (to override the "extra_repos: ..." YAML in the job definition).
+	//
+	// Technically we don't *need* this, so it may be desirable to just delete
+	// this field altogether for now from the proto (only add it when we need it
+	// in the future).
+	//extraRepos := req.GetExtraRepos()
+
+	return &refs, nil
+}
+
+type JobStruct struct {
+	Job *interface{}
+	// Encode the type of the Job to coerce into (via type assertion), so that
+	// users of JobStruct know how to make sense of the Job details.
+	JobExecutionType JobExecutionType
+}
+
+// We have to pay the cost of code duplication for Presubmits vs Postsubmits
+// because although they are different types they behave almost the same way for
+// purposes of fetching their job definitions. Either we duplicate a lot of the
+// same business logic, or we duplicate a lot of lower-level "infra" logic. We
+// choose to do the latter here. For an approach of duplicating the business
+// logic (which is less desirable), see the code for the "subscriber" package
+// used by the Sub component, specifically getProwJobSpec().
+type presubmitFetchers struct{}
+type postsubmitFetchers struct{}
+
+type jobFetchers interface {
+	getJobsFromStaticConfig(cfg *config.Config, orgRepo string) []interface{}
+	getJobsFromInRepoConfig(cfg *config.Config, pc *config.InRepoConfigCacheHandler, orgRepo string, baseSHAGetter config.RefGetter, headSHAGetters ...config.RefGetter) ([]interface{}, error)
+	couldRun(job interface{}, ref string) bool
+	name(job interface{}) string
+}
+
+func (jf *presubmitFetchers) getJobsFromStaticConfig(cfg *config.Config, orgRepo string) []interface{} {
+	jobs := cfg.GetPresubmitsStatic(orgRepo)
+	ret := make([]interface{}, len(jobs))
+	for i, v := range jobs {
+		ret[i] = v
+	}
+	return ret
+}
+
+func (jf *postsubmitFetchers) getJobsFromStaticConfig(cfg *config.Config, orgRepo string) []interface{} {
+	jobs := cfg.GetPostsubmitsStatic(orgRepo)
+	ret := make([]interface{}, len(jobs))
+	for i, v := range jobs {
+		ret[i] = v
+	}
+	return ret
+}
+
+func (jf *presubmitFetchers) couldRun(job interface{}, ref string) bool {
+	typedJob := job.(config.Presubmit)
+	return typedJob.CouldRun(ref)
+}
+
+func (jf *postsubmitFetchers) couldRun(job interface{}, ref string) bool {
+	typedJob := job.(config.Postsubmit)
+	return typedJob.CouldRun(ref)
+}
+
+func (jf *presubmitFetchers) name(job interface{}) string {
+	typedJob := job.(config.Presubmit)
+	return typedJob.Name
+}
+
+func (jf *postsubmitFetchers) name(job interface{}) string {
+	typedJob := job.(config.Postsubmit)
+	return typedJob.Name
+}
+
+func (jf *presubmitFetchers) getJobsFromInRepoConfig(cfg *config.Config, pc *config.InRepoConfigCacheHandler, orgRepo string, baseSHAGetter config.RefGetter, headSHAGetters ...config.RefGetter) ([]interface{}, error) {
```

Comment:
Note how this is different than https://github.com/kubernetes/test-infra/blob/b79005b5d4303e0ef0f35c9bbd6b66e40a6cb5ac/prow/config/config.go#L470. It's "purer" because it doesn't mix static configs with inrepoconfigs.

Reply:
This function has become obsolete in the course of addressing PR comments.